src/main/java/cn/nukkit/block/BlockDirt.java

```diff
@@ -54,10 +54,17 @@ public String getName() {
     @Override
     public boolean onActivate(Item item, Player player) {
         if (item.isHoe()) {
-            item.useOn(this);
-            this.getLevel().setBlock(this, this.getDamage() == 0 ? get(FARMLAND) : get(DIRT), true);
-
-            return true;
+            if (this.up() instanceof BlockAir) {
+                item.useOn(this);
+                this.getLevel().setBlock(this, this.getDamage() == 0 ? get(FARMLAND) : get(DIRT), true);
+                return true;
+            }
+        } else if (item.isShovel()) {
+            if (this.up() instanceof BlockAir) {
+                item.useOn(this);
+                this.getLevel().setBlock(this, Block.get(BlockID.GRASS_PATH));
```

Comment:
I would use ´get(GRASS_PATH)´ instead of ´Block.get(BlockID.GRASS_PATH)´

Reply:
already edited